### DISCUSSION TITLE
ScriptReference/MonoBehaviour.OnCollisionExit

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#489] [Notes]
# ScriptReference/MonoBehaviour.OnCollisionExit

https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnCollisionExit.html

**Comment by jschiff** [comment]
This callback will _not_ run if the collision ends for various reasons other than the objects physically moving apart, such as:
- One of the objects being destroyed, disabled, or deactivated
- A collider's shape being changed (e.g. changing the mesh on a MeshCollider, or changing the radius of a SphereCollider
